app/src/main/java/com/concordia/mcga/fragments/TransportButtonFragment.java

```diff
@@ -85,33 +85,50 @@ public void onClick(View clickedView) {
         toggle();
         switch (clickedView.getId()) {
             case R.id.transportExpandFAB:
-                if (!isExpanded())
+                if (!isExpanded()) {
                     restorePreviousIcon();
+                }
+                displayAllTransportTimes();
                 break;
             case R.id.walkFAB:
-                this.transportType = Transportation.WALK;
-                swapIcons(Transportation.WALK.getIconID());
+                this.transportType = TransportMode.WALKING;
+                swapIcons(Transportation.WALKING.getIconID());
                 break;
             case R.id.bikeFAB:
-                this.transportType = Transportation.BIKE;
-                swapIcons(Transportation.BIKE.getIconID());
+                this.transportType = TransportMode.BICYCLING;
+                swapIcons(Transportation.BICYCLING.getIconID());
                 break;
             case R.id.carFAB:
-                this.transportType = Transportation.CAR;
-                swapIcons(Transportation.CAR.getIconID());
+                this.transportType = TransportMode.DRIVING;
+                swapIcons(Transportation.DRIVING.getIconID());
                 break;
             case R.id.publicTransportFAB:
-                this.transportType = Transportation.PUBLIC_TRANSPORT;
-                swapIcons(Transportation.PUBLIC_TRANSPORT.getIconID());
+                this.transportType = TransportMode.TRANSIT;
+                swapIcons(Transportation.TRANSIT.getIconID());
                 break;
             case R.id.shuttleFAB:
-                this.transportType = Transportation.SHUTTLE;
+                this.transportType = "shuttle";
```

Comment:
Because I am using an external library which only uses the 4 types of transportation (Walk, Bicycle, Bus and Car). Now that i think of it, i can create a class that can wrap the external Library class. Will fix.